src/dist/replication/test/meta_test/unit_test/meta_split_service_test.cpp

```diff
@@ -56,9 +56,64 @@ class meta_split_service_test : public meta_test_base
         return rpc.response();
     }
 
+    register_child_response
+    register_child(ballot req_parent_ballot, ballot child_ballot, bool wait_zk = false)
+    {
+        // mock local app info
+        auto app = find_app(NAME);
+        app->partition_count *= 2;
+        app->partitions.resize(app->partition_count);
+        app->helpers->contexts.resize(app->partition_count);
+        for (int i = 0; i < app->partition_count; ++i) {
+            app->helpers->contexts[i].config_owner = &app->partitions[i];
+            app->partitions[i].pid = dsn::gpid(app->app_id, i);
+            if (i >= app->partition_count / 2) {
+                app->partitions[i].ballot = invalid_ballot;
+            } else {
+                app->partitions[i].ballot = PARENT_BALLOT;
+            }
+        }
+        app->partitions[CHILD_INDEX].ballot = child_ballot;
+
+        // mock node state
+        node_state node;
+        node.put_partition(dsn::gpid(app->app_id, PARENT_INDEX), true);
+        mock_node_state(dsn::rpc_address("127.0.0.1", 10086), node);
+
+        // mock register_child_request
+        partition_configuration parent_config;
+        parent_config.ballot = req_parent_ballot;
+        parent_config.last_committed_decree = 5;
+        parent_config.max_replica_count = 3;
+        parent_config.pid = dsn::gpid(app->app_id, PARENT_INDEX);
+
+        dsn::partition_configuration child_config;
+        child_config.ballot = PARENT_BALLOT + 1;
+        child_config.last_committed_decree = 5;
+        child_config.pid = dsn::gpid(app->app_id, CHILD_INDEX);
+
+        // register_child_request request;
+        auto request = dsn::make_unique<register_child_request>();
+        request->app.app_id = app->app_id;
+        request->parent_config = parent_config;
+        request->child_config = child_config;
+        request->primary_address = dsn::rpc_address("127.0.0.1", 10086);
+
+        register_child_rpc rpc(std::move(request), RPC_CM_REGISTER_CHILD_REPLICA);
+        split_svc().register_child_on_meta(rpc);
+        wait_all();
+        if (wait_zk) {
```

Comment:
what's `wait_zk` used for? `wait_all` supposed to wait any tasks to complete.

Reply:
`wait_all` function will wait meta_service tasks, zk tasks are not included.